src/Aspire.Hosting.Sdk/SDK/Sdk.targets

```diff
@@ -20,7 +18,21 @@
     <IsPackable Condition="'$(IsPackable)' == ''">false</IsPackable>
   </PropertyGroup>
 
-  <!-- *** END ***  -->
+  <!--
+  Default AppHost ProjectReference items to not be referenced in the AppHost, unless it is AppHostLibrary=true.
+  This defaulting needs to happen in the SDK targets so this metadata affects NuGet Restore.
+  -->
+  <ItemGroup Condition="'$(IsAspireHost)' == 'true'">
+    
+    <ProjectReference Update="@(ProjectReference)">
```

Comment:
Should we add a global opt-out property for this behavior? e.g. `EnableAspireProjectReferenceProcessing`

Reply:
Can you think of a good reason why?

Reply:
Still slightly uneasy about taking over ProjectReferences and it's common for MSBuild defaults to have an opt-out. But we can wait for feedback.